fetcher.go

```diff
@@ -71,6 +72,11 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 	if mime == "application/vnd.ipld.raw" {
 		resourceType = "block"
 	}
+	fetchCalledTotalMetric.WithLabelValues(resourceType).Add(1)
+	if ctx.Err() != nil {
+		fetchRequestContextErrorTotalMetric.WithLabelValues(resourceType, ctx.Err().Error(), "init").Add(1)
```

Comment:
:warning:   @aarshkshah1992 @willscott  this is potentially really really bad, because you have unbounded number of values `.Error()`   could return, which will in turn baloon the size of metrics.

Last time I checked we had errors which had the URL / CID of failed request, and that means every error will be unique, exploding metrics space:


```
http request failed: Get \"https://64.44.166.190/ipfs/bafybeiaysi4s6lnjev27ln5icwm6tueaw2vdykrtjkwiphwekaywqhcjze/favicon.ico?format=car&depth=0\": context canceled
```

Reply:
https://github.com/filecoin-saturn/caboose/pull/81

Reply:
@lidel Thanks for catching this ! In my mind, the context cancelled error could only be Cancelled or Dead line exceeded ! But I see now.